lib/sync/db-templates/materializedViews.template.sql

```diff
@@ -0,0 +1,190 @@
+			AND type = 'StudentEnrollment'
+		GROUP BY user_id, course_id, type
+	),
+	/*
+	 * Filter requests where user role in enrollment is facutly
+	 * TODO: Add page_view and participation dictionary tables to further filter requests
+	 */
+	 w0 AS (
+	 	SELECT
+			r.id,
+			r.timestamp,
+			r.timestamp_year,
+			r.timestamp_month,
+			r.timestamp_day,
+			r.user_id,
+			r.course_id,
+			r.root_account_id,
+			r.course_account_id,
+			r.quiz_id,
+			r.discussion_id,
+			r.conversation_id,
+			r.assignment_id,
+			r.url,
+			r.user_agent,
+			r.http_method,
+			r.remote_ip,
+			r.interaction_micros,
+			r.web_application_controller,
+			r.web_application_action,
+			r.web_application_context_type,
+			r.web_application_context_id,
+			r.real_user_id,
+			r.session_id,
+			r.user_agent_id,
+			r.http_status,
+			r.http_version,
+			e1.type AS enrollment_type
+		FROM <%= externalSchema %>.requests r
+			INNER JOIN e1
+				ON r.user_id = e1.user_id
+					AND r.course_id = e1.course_id
+		WHERE e1.type = 'StudentEnrollment'
+					AND r.user_id is not null
+					AND r.course_id is not null
+	 ),
+	 w1 AS (
+	 	-- Calculates activity frequency for a user on a course level
+		SELECT
+			course_id,
+			user_id,
+			COALESCE(count(url),0) AS user_page_views
+		FROM w0
+		WHERE course_id IS NOT NULL
+			AND user_id IS NOT NULL
+		GROUP BY course_id, user_id
+	),
+	/*
+	 * Calculates activity frequency for each course
+	 * TODO: There are a few courses that span enrollment terms
+	 * Check if the frequency is correctly calculated for them
+	 */
+	 w2 AS (
+		SELECT
+			course_id,
+			COALESCE(AVG(user_page_views),0) AS avg_course_page_views,
+			CAST(STDDEV_POP(user_page_views) AS dec(14,2)) stddev_course_page_views
+		FROM w1
+		GROUP BY course_id
+	),
+	w3 AS (
+		-- Calculates Z-scores for users records having non zero std dev
+		SELECT
+			w1.course_id AS course_id,
+	   		w1.user_id AS user_id,
+	   		w1.user_page_views,
+	   		w2.avg_course_page_views,
+	        w2.stddev_course_page_views,
+	   		(w1.user_page_views - w2.avg_course_page_views)/(w2.stddev_course_page_views) AS user_page_view_zscore
+		FROM w1 JOIN w2
+		ON w1.course_id = w2.course_id
+		WHERE w2.stddev_course_page_views > 0.00
+	),
+	w4 AS (
+		-- page_views_zscore calculation query, where nulls/divide by zero errors are handled
+		SELECT
+			w1.course_id AS course_id,
+			w1.user_id AS user_id,
+		   	w1.user_page_views,
+	   	  	w2.avg_course_page_views,
+	       	w2.stddev_course_page_views,
+	   	   	COALESCE(w3.user_page_view_zscore,0) AS user_page_view_zscore
+		FROM w1
+			LEFT JOIN w3
+				ON w1.course_id = w3.course_id AND w1.user_id = w3.user_id
+			LEFT JOIN w2
+				ON w1.course_id = w2.course_id
+	),
+	/*
+	 * Combines user_dim and pseudonym_dim tables to retrieve correct
+	 * global_user_id, canavs_user_id, sis_login_id and sis_user_id
+	 * There are multiple entries in the table which maps to canvas_user_ids.
+	 * This occurs when
```

Comment:
Interrupted sentence. Also, shouldn't the previous sentence be "There _may_ be multiple entries in the table for a single canvas_user_id."?